runner_service/controllers/utils.py

```diff
@@ -17,24 +18,32 @@ def requires_auth(f):
     def decorated(*args, **kwargs):
         """ check the request carries a valid username/password header """
 
-        # # check credentials supplied in the http request are valid
-        # auth = request.authorization
-        # if not auth:
-        #     return jsonify(message="Missing credentials"), 401
-        #
-        # if (auth.username != settings.config.api_user or
-        #    auth.password != settings.config.api_password):
-        #     return jsonify(message="username/password mismatch with the "
-        #                            "configuration file"), 401
-
         # if there is a whitelist and if request came from not whitelisted ip
         if configuration.settings.ip_whitelist and request.remote_addr not in configuration.settings.ip_whitelist:
             response = APIResponse()
             response.status, response.msg = "NOAUTH", "Access denied not on whitelist"
             logger.info("{} made a request and is not whitelisted".format(request.remote_addr))
             return response.__dict__, BaseResource.state_to_http[response.status]
         else:  # there is no whitelist let everything through or it came from a whitelisted ip
-            return f(*args, **kwargs)
+            # if login move on
+            if request.path == "/api/v1/login":
+                return f(*args, **kwargs)
+            else:  # check for valid token
+                token = request.headers.get('Authorization')
+                try:
+                    jwt.decode(token, configuration.settings.token_secret, algorithms='HS256')
+                except jwt.DecodeError:
+                    response = APIResponse()
+                    response.status, response.msg = "NOAUTH", "Access denied invalid token"
```

Comment:
If I don't get a token first I see this error invalid token. Can we check for no token and change the error response to help guide the dev?

Reply:
+1